File: src/B_10773/B_10773.java
package B_10773;

import java.io.*;
import java.util.LinkedList;
import java.util.Stack;

public class B_10773 {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        Stack<Integer> stack = new Stack<>();
        int res=0,n=Integer.parseInt(br.readLine());
        for(int i=0; i< n;i++){
            int in = Integer.parseInt(br.readLine());
            if(in == 0){
                res-=stack.pop();
            }
            else{
                stack.push(in);
                res+=in;
            }
        }
        System.out.println(res);
    }
}

File: src/B_10773/B_10773.kt
package B_10773

import java.io.BufferedReader
import java.io.InputStreamReader
import java.util.*



fun main(args: Array<String>) {
    val br = BufferedReader(InputStreamReader(System.`in`))
    val stack = Stack<Int>()
    var res = 0
    val n = br.readLine().toInt()
    for (i in 0 until n) {
        val `in` = br.readLine().toInt()
        if (`in` == 0) {
            res -= stack.pop()
        } else {
            stack.push(`in`)
            res += `in`
        }
    }
    println(res)
}
